src/main/java/mate/academy/spring/validation/EmailValidation.java

```diff
@@ -0,0 +1,20 @@
+package mate.academy.spring.validation;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class EmailValidation implements ConstraintValidator<EmailConstraint, String> {
+    private static final String EMAIL_PATTERN = "^[A-Z\\d._%+-]+@[A-Z\\d.-]+\\.[A-Z]{2,6}$";
+
+    @Override
+    public boolean isValid(String email, ConstraintValidatorContext context) {
+        if (email == null) {
+            return false;
+        }
+        Pattern pattern = Pattern.compile(EMAIL_PATTERN, Pattern.CASE_INSENSITIVE);
```

Comment:
It is a good practice to compile patterns in only once in such services. Let's better make a constant of a type Pattern but not String